java-package/openbci_gui_helpers/src/main/java/openbci_gui_helpers/GUIHelper.java

```diff
@@ -47,23 +56,53 @@ private interface DllNativeInterface extends Library
         String lib_path = unpack_from_jar (lib_name);
         String lib_native_path = unpack_from_jar (lib_native_name);
 
+        /*
+        if (SystemUtils.IS_OS_MAC) {
+            String path = GUIHelper.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+            File jarFile = new File(path);
+            String jarDir = jarFile.getParentFile().getAbsolutePath();
+            lib_path = jarDir + File.separator + lib_name;
+            lib_native_path = jarDir + File.separator + lib_native_name;
+            System.out.println("jarDir: " + jarDir);
+            System.out.println("lib_path: " + lib_path);
+            System.out.println("lib_native_path: " + lib_native_path);
+        }
+        */
+
         instance = (DllInterface) Native.load (lib_path, DllInterface.class);
         instance_native = (DllNativeInterface) Native.load (lib_native_path, DllNativeInterface.class);
     }
 
     private static String unpack_from_jar (String lib_name)
     {
+        File file = null;
+        InputStream link = null;
         try
         {
-            File file = new File (lib_name);
+            file = new File (lib_name);
             if (file.exists ())
                 file.delete ();
-            InputStream link = (GUIHelper.class.getResourceAsStream (lib_name));
-            Files.copy (link, file.getAbsoluteFile ().toPath ());
+            link = (GUIHelper.class.getResourceAsStream (lib_name));
+            if (SystemUtils.IS_OS_MAC) {
+                String jarPath = GUIHelper.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+                File jarFile = new File(jarPath);
+                String libPathString = jarFile.getParentFile().getAbsolutePath() + File.separator + lib_name;
+                Path libPath = Paths.get(libPathString);
+                Files.copy (link, libPath);
+                return libPathString;
+            } else {
```

Comment:
Info - We were having problems unpacking on MacOS before. What ended up fixing it?

Reply:
This uses a new approach to unpack the files, and only attempts this on Mac.

Two scenarios:
- this attempt works
- we just put the .dylibs in the `library` folder alongside the `.jar` file (aka pre-unpacked)

Probably requires a little more testing. Though the second scenario is already used in  https://github.com/OpenBCI/OpenBCI_GUI/releases/tag/v5.2.2 on the Mac release only. This build was created locally and also signed.

Reply:
Testing has been requested from other team members. Leaving this convo unresolved for now.

Reply:
I'd like to leave both attempts in. No harm in doing so.